@xen-orchestra/fs/src/abstract.js

```diff
@@ -284,15 +284,23 @@ export default class RemoteHandlerAbstract {
     return this._encryptor.decryptData(data)
   }
 
-  async rename(oldPath, newPath, { checksum = false } = {}) {
+  async rename(oldPath, newPath, { checksum = false, createTree = true } = {}) {
     oldPath = normalizePath(oldPath)
     newPath = normalizePath(newPath)
-
-    let p = timeout.call(this._rename(oldPath, newPath), this._timeout)
-    if (checksum) {
-      p = Promise.all([p, this._rename(checksumFile(oldPath), checksumFile(newPath))])
+    try{
+      let p = timeout.call(this._rename(oldPath, newPath), this._timeout)
+      if (checksum) {
+        p = Promise.all([p, this._rename(checksumFile(oldPath), checksumFile(newPath))])
+      }
+      await p
+    }catch(error){
+      // ENOENT can be a missing target directory OR a missing source
+      if(error.code === 'ENOENT' && createTree){
+        await this.mktree(dirname(newPath))
+        return this.rename(oldPath, newPath, {checksum, createTree: false})
```

Comment:
Avoid calling public methods, they are decorated (i.e. special behaviors are added, e.g. prefixes, concurrency and timeout) which can lead to surprises.